WalletWasabi.Backend/Controllers/SoftwareController.cs

```diff
@@ -10,16 +14,30 @@ namespace WalletWasabi.Backend.Controllers
 	[Route("api/[controller]")]
 	public class SoftwareController : Controller
 	{
+		public VersionsResponse VersionsResponse { get; private set; }
+
 		/// <summary>
 		/// Gets the latest versions of the client and backend.
 		/// </summary>
 		/// <returns>ClientVersion, BackendMajorVersion.</returns>
 		/// <response code="200">ClientVersion, BackendMajorVersion.</response>
 		[HttpGet("versions")]
 		[ProducesResponseType(typeof(VersionsResponse), 200)]
-		public VersionsResponse GetVersions()
+		public async Task<IActionResult> GetVersionsAsync()
 		{
-			return Helpers.Constants.VersionsResponse;
+			if (VersionsResponse is null)
+			{
+				VersionsResponse = new VersionsResponse()
+				{
+					ClientVersion = Constants.ClientVersion.ToString(3),
```

Comment:
Use `ToVersionString()` instead of `ToString(3)`
`ClientVersion = Constants.ClientVersion.ToVersionString(),`

Reply:
Here we want to use `ToString(3)` actually, the backend shouldn't know if you're using a debug version or not, only the user should see that in their own client.